pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -315,6 +318,20 @@ func (config *V2RegistriesConf) postProcess() error {
 		config.UnqualifiedSearchRegistries[i] = registry
 	}
 
+	// Registries are ordered and the first longest prefix always wins,
+	// rendering later items with the same prefix non-existent. We cannot error
+	// out anymore as this might break existing users, so let's just ignore them
+	// to guarantee that the same prefix exists only once.
+	prefixMap := make(map[string]bool)
+	uniqueRegistries := []Registry{}
+	for i := range config.Registries {
+		if _, exists := prefixMap[config.Registries[i].Prefix]; exists {
+			continue
```

Comment:
Non-blocking: should this warn? I suppose that would be a separate PR.

(Absolutely non-blocking: just using `if …; !exists` without a `continue` might be tiny bit easier to read, maybe.)

Reply:
I added a TODO and changed the code as suggested.